### PR TITLE
fix cmake compile - no need to link to lsockets on macOS

### LINKED ISSUE #11887: cmake: macOS not compiling missing -lsocket in linking stage - Ventura on M1
###
Not compiling
Linking failing when compiling for macOS with latest 8.2.0 with not finding socket:
-lsocket

```  
 cd "build_${TYPE}_${PLATFORM}"
        cmake  .. \
            -DCMAKE_C_STANDARD=17 \
            -DCMAKE_CXX_STANDARD=17 \
            -DCMAKE_CXX_STANDARD_REQUIRED=ON \
            -DCMAKE_CXX_FLAGS="-DUSE_PTHREADS=1" \
            -DCMAKE_C_FLAGS="-DUSE_PTHREADS=1" \
            -DCMAKE_CXX_EXTENSIONS=OFF \
            -DBUILD_SHARED_LIBS=OFF \
            -DCMAKE_BUILD_TYPE=Release \
            -DCMAKE_INSTALL_PREFIX=Release \
            -DCMAKE_INCLUDE_OUTPUT_DIRECTORY=include \
            -DCMAKE_INSTALL_INCLUDEDIR=include \
            -DCMAKE_TOOLCHAIN_FILE=$APOTHECARY_DIR/ios.toolchain.cmake \
            -DPLATFORM=$PLATFORM \
            -DENABLE_BITCODE=OFF \
            -DENABLE_ARC=OFF \
            -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE \
            -DENABLE_VISIBILITY=OFF \
            -DOPENSSL_ROOT_DIR="$LIBS_ROOT/openssl/" \
            -DOPENSSL_INCLUDE_DIR="$LIBS_ROOT/openssl/include" \
            -DOPENSSL_SSL_LIBRARY="$LIBS_ROOT/openssl/lib/$TYPE/$PLATFORM/libssl.a" \
            -DOPENSSL_CRYPTO_LIBRARY="$LIBS_ROOT/openssl/lib/$TYPE/$PLATFORM/libcrypto.a" \
            -DENABLE_UNIX_SOCKETS=ON \
            -DCURL_ENABLE_SSL=ON \
            -DHTTP_ONLY=ON \
            -DCMAKE_MACOSX_BUNDLE=OFF \
            -DUSE_SECURE_TRANSPORT=ON \
            -DUSE_NGHTTP2=OFF \
            -DUSE_LIBIDN2=OFF \
            -DENABLE_LDAP=OFF \
            -DENABLE_LDAPS=OFF \
            -DENABLE_VERBOSE=ON \
            -DENABLE_THREADED_RESOLVER=OFF \
            -DENABLE_IPV6=ON \
            -G Xcode
        cmake --build . --config Release --target install
        cd ..
```

Error: ld: library 'socket' not found

As apart of the Linking stage:
```
Ld /Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/src/Release/curl normal (in target 'curl' from project 'CURL')
    cd /Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Xlinker -reproducible -target x86_64-apple-macos14.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -O3 -L/Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/build/EagerLinkingTBDs/Release -L/Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/src/Release -F/Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/build/EagerLinkingTBDs/Release -F/Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/src/Release -filelist /Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/build/curl.build/Release/Objects-normal/x86_64/curl.LinkFileList -Wl,-search_paths_first -Wl,-headerpad_max_install_names /Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/lib/Release/libcurl.a -lsocket -framework SystemConfiguration -framework CoreFoundation /Users/curlbuildplz/SOURCE/apothecary/openssl/lib/osx/MAC/libssl.a /Users/curlbuildplz/SOURCE/apothecary/openssl/lib/osx/MAC/libcrypto.a /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/lib/libz.tbd -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker /Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/build/curl.build/Release/Objects-normal/x86_64/curl_dependency_info.dat -o /Users/curlbuildplz/SOURCE/apothecary/apothecary/build/curl/build_osx_MAC/src/Release/curl
```

-lsocket does not seem to be here or or a known system path anymore 

		

### I expected the following

Library Built for macOS 

### curl/libcurl version

curl 8.2.0

### operating system

macOS 13.5.2 Ventura - M1X Macbook - arm64

### PR DESCRIPTION
**Pull request by danoli3**
Fixes #11887 

Added NOT APPLE to not add library socket  to linker on macOS

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1096,7 +1098,7 @@ endif()
 # Check for some functions that are used
 if(WIN32)
   set(CMAKE_REQUIRED_LIBRARIES ws2_32)
-elseif(HAVE_LIBSOCKET)
+elseif(HAVE_LIBSOCKET AND NOT APPLE)
```
Comment (bagder) [inline CMakeLists.txt:1101]: On apple, surely `HAVE_LIBSOCKET` would never be set so why add an extra variable to check here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -376,7 +376,9 @@ if(ENABLE_THREADED_RESOLVER)
 endif()
 
 # Check for all needed libraries
+if(NOT APPLE)
 check_library_exists_concat("socket" connect      HAVE_LIBSOCKET)
+endif()
```
Comment (bagder) [inline CMakeLists.txt:381]: But why? For speed reasons?

File: CMakeLists.txt (hunk 1 above)
Comment (danoli3) [reply CMakeLists.txt:1101]: it failed to compile without this

Comment (jay) [reply CMakeLists.txt:1101]: in what way did it fail to compile, because you are the first to report this and it has been there for a decade

**Comment by bagder** [comment]
Since #11887 is now closed, I figure we can close this one as well?

**Comment by danoli3** [comment]
Yeah basically don't ever do G

**Pull request closed without merging**